public/app/features/plugins/datasource_srv.ts

```diff
@@ -17,7 +17,7 @@ import { GrafanaRootScope } from 'app/routes/GrafanaCtrl';
 import { expressionDatasource } from 'app/features/expressions/ExpressionDatasource';
 
 export class DatasourceSrv implements DataSourceService {
-  datasources: { [name: string]: DataSourceApi };
+  datasources: Record<string, DataSourceApi>;
```

Comment:
Hm, never seen `Record` before. Is this a new thing introduced in TS3.7? We also have wildly used  `KeyValue` type in grafana/data which could probably be replaced with `Record`

Reply:
It's existed for a while as far as I know (https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt). Mostly a personal preference thing but I think it reads a bit better than `{ [key: string]: DataSourceApi }`